services/outputhost/extcache.go

```diff
@@ -237,15 +236,28 @@ func (extCache *extentCache) loadReplicaStream(startAddress int64, startSequence
 
 		// First try to start from the already set offset in metadata
 		if startAddress == 0 {
-			// If consumer group wants to start from a timestamp, get the address from the store
-			startFrom := extCache.startFrom
+			// If consumer group wants to start from a timestamp, get the address from the store.
+			// Note: we take into account any 'delay' that is associated with the CG and the 'skip-older'
+			// time that was specified, by offsetting the time appropriately. we apply the 'start-from'
+			// and 'skip-older' to the "visibility time" (enqueue-time + delay) of the message as opposed
```

Comment:
I'm not sure I agree with applying the delay to start-from...

If I were to say 'start-from = tuesday', delay = 48h, then I expect to receive the first message enqueued on Tuesday at 12:00:01am on Thursday.

Doesn't seem like you need to take delay into account in start-from for that to work out.